source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,346 @@ Sorts Builders
+   collection.find().sort(orderBySort);
+
+The above code snippet returns the documents in the
+:ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`
+collection in the following order: 
+
+.. code-block:: json
+
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+
+Text Score
+----------
+
+If you are using :manual:`$text </reference/operator/query/text/>`
+queries to search your data you can sort the results of those
+queries by text score (how well your search matched the returned document).
+To sort your results by the
+:manual:`$meta </reference/operator/aggregation/meta/#exp._S_meta>`
+``textScore`` associated with a ``$text`` query, use the
+``Sorts.metaTextScore()`` static factory method to retrieve a sort specification
+for a descending sort by score (best match first).
+
+.. note::
+
+  If your MongoDB instance is running MongoDB 4.2 and earlier,
+  to use a ``Sorts.metaTextScore()``
+  expression with ``FindIterable.sort()``, you must also
+  include the same expression in a projection. This is demonstrated
+  in the code example below.
+
+The ``$text`` command performs a text search on the content indexed by the 
+:manual:`text index </core/index-text/>` of your collection. You must create
+a text index on your collection before you can run a ``$text`` query and use
+``Sorts.metaTextScore()``.
+
+.. example::
+
+   In the following code example, we show how to use the
+   ``metaTextScore()`` method to sort the results of a ``$text``
+   query on
+   :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`.
+   The following code snippet works as follows:
+
+   #. Create a text index for
+      :ref:`sort_example <docs_java_fundamentals_builders_sort_sort_example>`
+      on the ``food`` field.
+   #. Run a ``$text`` search for the phrase "maple donut".
+   #. Project ``metaTextScore()`` into your query results as the
+      ``score`` field. This step is optional if your MongoDB instance is
+      running MongoDB 4.4 or later. 
+   #. Sort using ``metaTextScore()`` as a sort specification.
+
+   .. code-block:: java
+   
+      import static com.mongodb.client.model.Sorts.metaTextScore;
+      import static com.mongodb.client.model.Sorts.ascending;
+   
+      // <MongoCollection setup code here>
+      
+      // create a text index on the food field
+      collection.createIndex(Indexes.text("food"));
+      // create text score sort specification
+      Bson metaTextScoreSort = metaTextScore("score");
+      String searchTerm = "maple donut";
+      // build $text search query
+      Bson searchQuery = new Document("$text", new Document("$search", searchTerm));
```

Comment:
Are you able to use the Text builder here? We can avoid mixing builders and Document ejson if so.

Reply:
Thanks for pointing this out. Had not yet taken a close look at the `Filters` class and didn't realize this was an option. Fix: https://github.com/mongodb/docs-java/pull/59/commits/261c95fdf997e6ac5d8f1d2d3634ec11686271d4